src/me/xdrop/fuzzywuzzy/algorithms/BasicAlgorithm.java

```diff
@@ -1,43 +1,43 @@
 package me.xdrop.fuzzywuzzy.algorithms;
 
 import me.xdrop.fuzzywuzzy.Applicable;
-import me.xdrop.fuzzywuzzy.StringProcessor;
+import me.xdrop.fuzzywuzzy.ToStringFunction;
 
 public abstract class BasicAlgorithm implements Applicable {
 
-    private StringProcessor stringProcessor;
+    private ToStringFunction<String> stringFunction;
 
     public BasicAlgorithm() {
-        this.stringProcessor = new DefaultStringProcessor();
+        this.stringFunction = new DefaultStringFunction();
     }
 
-    public BasicAlgorithm(StringProcessor stringProcessor) {
-        this.stringProcessor = stringProcessor;
+    public BasicAlgorithm(ToStringFunction<String> stringFunction) {
+        this.stringFunction = stringFunction;
     }
 
-    public abstract int apply(String s1, String s2, StringProcessor stringProcessor);
+    public abstract int apply(String s1, String s2, ToStringFunction<String> stringProcessor);
 
     public int apply(String s1, String s2){
 
-        return apply(s1, s2, this.stringProcessor);
+        return apply(s1, s2, this.stringFunction);
 
     }
 
-    public BasicAlgorithm with(StringProcessor stringProcessor){
-        setStringProcessor(stringProcessor);
+    public BasicAlgorithm with(ToStringFunction<String> stringFunction){
+        setStringFunction(stringFunction);
         return this;
     }
 
     public BasicAlgorithm noProcessor(){
-        this.stringProcessor = new NoProcess();
+        this.stringFunction = ToStringFunction.DEFAULT;
```

Comment:
This gets replaced by `DEFAULT` which I believe is not the same as `NoProcess`, the basic ratios should not have any processing done to them by default, so we'd still need to have something that represents an identity string function `input => input` for this use case.

Reply:
`NoProcess` just returned the unhandled input String, which is exactly what `DEFAULT` does, so I fail to see a difference. 🤔 https://github.com/xdrop/fuzzywuzzy/pull/47/files#diff-a5cadeadf032c3e499ee1134add85c93L9

Reply:
Maybe you got confused between `ToStringFunction.DEFAULT` and `DefaultStringFunction`?
The naming is odd on those ones. 
What could be a better name for the final DEFAULT variable?

Reply:
Ahh my bad yeah got confused between those two. Maybe we should call it `ToStringFunction.NO_PROCESS` or `ToStringFunction.IDENTITY`?

Reply:
i would go for "NO_PROCESS". will be contained in my next PR